pkg/kube/util.go

```diff
@@ -146,3 +156,120 @@ func CheckPodReady(pod *kubeApiCore.Pod) error {
 		return fmt.Errorf("%s", pod.Status.Phase)
 	}
 }
+
+// ClientConfig first tries to get the config from inside a kubernetes environment
+// If unsuccessful, it tries to get the config from the default kubeconfig file
+func ClientConfig() (*rest.Config, error) {
+	config, err := rest.InClusterConfig()
+	var ocErr error
+	if err != nil {
+		config, ocErr = clientcmd.BuildConfigFromFlags("", defaultKubeconfigPath())
+		if ocErr != nil {
+			return nil, fmt.Errorf("in-cluster error: %v, out-of-cluster error: %v", err, ocErr)
+		}
+	}
+	return config, nil
+}
+
+// Returns a kubernetes clientset with the default config
+// TODO: align and remove redundancy with BuildClientConfig, CreateClientset
+// as well as istio.io/istio/operator/cmd/mesh/shared.go, and potentially istio.io/istio/cni/cmd/istio-cni/kubernetes.go
+func Config() (*rest.Config, *kubernetes.Clientset, error) {
+	config, err := ClientConfig()
+	if err != nil {
+		return nil, nil, err
+	}
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return nil, nil, err
+	}
+	return config, clientset, nil
+}
+
+// Returns a kubernetes service in a namespace
+// TODO: remove once merged
+// nolint: deadcode
+func Service(clientset *kubernetes.Clientset, name, namespace string) (*kubeApiCore.Service, error) {
+	service, err := clientset.CoreV1().Services(namespace).Get(context.Background(), name, metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+	return service, nil
+}
+
+// Returns a kubernetes pod in a namespace
+func Pod(clientset *kubernetes.Clientset, name, namespace string) (*kubeApiCore.Pod, error) {
+	pods, err := clientset.CoreV1().Pods(namespace).List(context.Background(), metav1.ListOptions{})
+	if err != nil {
+		return nil, err
+	}
+	for _, pod := range pods.Items {
+		if strings.Contains(pod.Name, name) {
+			return &pod, nil
+		}
+	}
+	return nil, fmt.Errorf("pod with name %s not found in namespace %s", name, namespace)
+}
+
+// Returns a kubernetes config map in a namespace
+// TODO: remove once merged
+// nolint: deadcode
+func ConfigMap(clientset *kubernetes.Clientset, name, namespace string) (*kubeApiCore.ConfigMap, error) {
+	cm, err := clientset.CoreV1().ConfigMaps(namespace).Get(context.Background(), name, metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+	return cm, nil
+}
+
+// Execs a command inside a kubernetes pod and returns stdout and stderr
+// nolint: unparam
+func PodExec(config *rest.Config, clientset *kubernetes.Clientset, name, namespace, command string, stdin io.Reader) (sout, serr string, err error) {
```

Comment:
Use `kube.NewExtendedClient` (removes the need for Config and ClientConfig), then call `client.PodExec`. Please don't reimplement this